pkg/test/framework/api/components/apps.go

```diff
@@ -43,35 +43,30 @@ type Apps interface {
 	GetAppOrFail(name string, t testing.TB) App
 }
 
+// Service represents a deployed service within the mesh.
+type Service interface {
+	ClusterIP() string
```

Comment:
This shouldn't be in the interface since it's not appropriate/useful on all platforms. If you want to expose this from your k8s-specific implementation, that's fine.

Reply:
For me it makes sense to have access to the ip address of an application. Perhaps the name is misleading, because a ClusterIP is only used in kubernetes environments. In my view, the return of "127.0.0.1" for native environments is an appropriate value.

Reply:
I guess my question is: why do we need it? Keeping the API surface area as small as possible is a goal. If we expose it, developers are going to try to use it ... and it may or may not work as expected on a given platform.